Libraries/JavaScriptAppEngine/Initialization/InitializeJavaScriptAppEngine.js

```diff
@@ -70,6 +70,15 @@ function polyfillGlobal(name, newValue, scope = GLOBAL) {
   Object.defineProperty(scope, name, {...descriptor, value: newValue});
 }
 
+/**
+ * Polyfill a module if it is not already defined in `scope`.
+ */
+function polyfillIfNeeded(name, polyfill, scope = GLOBAL) {
+  if (typeof scope[name] === 'undefined') {
+    Object.defineProperty(scope, name, {value: polyfill});
```

Comment:
I wonder whether this is generally safe. In browsers, many properties are enumerable/writable/configurable, e.g. `navigator`.